pkg/cmd/adm/restart.go

```diff
@@ -1,157 +1,213 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	kubectlrollout "k8s.io/kubectl/pkg/cmd/rollout"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
 	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+type NonOperatorDeploymentsRestarterFunc func(ctx *clicontext.CommandContext, deployment appsv1.Deployment, f cmdutil.Factory, ioStreams genericclioptions.IOStreams) error
+
+type RolloutStatusCheckerFunc func(ctx *clicontext.CommandContext, f cmdutil.Factory, ioStreams genericclioptions.IOStreams, labelSelector string) error
+
+// NewRestartCmd() is a function to restart the whole operator, it relies on the target cluster and fetches the cluster config
+// 1.  If the command is run for host operator, it restart the whole host operator.(it deletes olm based pods(host-operator pods),
+// waits for the new pods to come up, then uses rollout-restart command for non-olm based - registration-service)
+// 2.  If the command is run for member operator, it restart the whole member operator.(it deletes olm based pods(member-operator pods),
+// waits for the new pods to come up, then uses rollout-restart command for non-olm based deployments - webhooks)
 func NewRestartCmd() *cobra.Command {
-	var targetCluster string
 	command := &cobra.Command{
-		Use:   "restart -t <cluster-name> <deployment-name>",
-		Short: "Restarts a deployment",
-		Long: `Restarts the deployment with the given name in the operator namespace. 
-If no deployment name is provided, then it lists all existing deployments in the namespace.`,
-		Args: cobra.RangeArgs(0, 1),
+		Use:   "restart <cluster-name>",
+		Short: "Restarts an operator",
+		Long: `Restarts the whole operator, it relies on the target cluster and fetches the cluster config
+		1.  If the command is run for host operator, it restarts the whole host operator.
+		(it deletes olm based pods(host-operator pods),waits for the new pods to 
+		come up, then uses rollout-restart command for non-olm based deployments - registration-service)
+		2.  If the command is run for member operator, it restarts the whole member operator.
+		(it deletes olm based pods(member-operator pods),waits for the new pods 
+		to come up, then uses rollout-restart command for non-olm based deployments - webhooks)`,
+		Args: cobra.ExactArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
 			term := ioutils.NewTerminal(cmd.InOrStdin, cmd.OutOrStdout)
 			ctx := clicontext.NewCommandContext(term, client.DefaultNewClient)
-			return restart(ctx, targetCluster, args...)
+			return restart(ctx, args...)
 		},
 	}
-	command.Flags().StringVarP(&targetCluster, "target-cluster", "t", "", "The target cluster")
-	flags.MustMarkRequired(command, "target-cluster")
 	return command
 }
 
-func restart(ctx *clicontext.CommandContext, clusterName string, deployments ...string) error {
+func restart(ctx *clicontext.CommandContext, clusterNames ...string) error {
+	clusterName := clusterNames[0]
```

Comment:
there should be only one clusterName provided, there is no need to support more of them
```suggestion
func restart(ctx *clicontext.CommandContext, clusterName string) error {
```

Reply:
We are only supporting 1 argument https://github.com/kubesaw/ksctl/pull/79/files/2704a913cdfcff86a6bcd36433d2b14887d19f24#diff-281b31e7723a650782511ee228ab82b4065dd935ce2d4e4c0fb0f491ff6732bfR41
 but my understanding was cobra command needs a list and hence just a usage

Reply:
but you can call by passing the first value from the slice
```go
restart(ctx, args[0])
```

Reply:
Oh okay, i understood it differently, changed it now